code/utilities/helpers/EnvHelper.py

```diff
@@ -40,13 +42,17 @@ def __init__(self, **kwargs) -> None:
         self.AZURE_OPENAI_STREAM = os.getenv('AZURE_OPENAI_STREAM', '')
         self.AZURE_OPENAI_EMBEDDING_MODEL = os.getenv('AZURE_OPENAI_EMBEDDING_MODEL', '')
          # Set env for OpenAI SDK
-        self.OPENAI_API_TYPE = "azure"
         self.OPENAI_API_BASE = f"https://{os.getenv('AZURE_OPENAI_RESOURCE')}.openai.azure.com/"
-        self.OPENAI_API_KEY = self.AZURE_OPENAI_KEY
+        if self.AZURE_AUTH_TYPE == "keys": 
+            self.OPENAI_API_KEY = self.AZURE_OPENAI_KEY
+            self.OPENAI_API_TYPE = "azure"
+        else:
+            self.OPENAI_API_KEY = DefaultAzureCredential(exclude_shared_token_cache_credential=True).get_token("https://cognitiveservices.azure.com/.default").token
+            self.OPENAI_API_TYPE = "azure_ad"
         self.OPENAI_API_VERSION = self.AZURE_OPENAI_API_VERSION
-        os.environ["OPENAI_API_TYPE"] = "azure"
+        os.environ["OPENAI_API_TYPE"] = self.OPENAI_API_TYPE
         os.environ["OPENAI_API_BASE"] = f"https://{os.getenv('AZURE_OPENAI_RESOURCE')}.openai.azure.com/"
-        os.environ["OPENAI_API_KEY"] = self.AZURE_OPENAI_KEY
+        os.environ["OPENAI_API_KEY"] = self.OPENAI_API_KEY
```

Comment:
the openai_api_key is also being set in the environ variables, so if we move the logic to LLMHelper constructor, then we need to set the environ variable os.environ["OPENAI_API_KEY"] as well which I am guessing can also become confusing because all other variables are being set here in the EnvHelper file. What do you think? @adamdougal

Reply:
Yea that's a very good point. I'm going to backtrack and say that how you have it already is best, keeping the logic in one place!